pkg/controllers/user/logging/utils/logging.go

```diff
@@ -10,22 +10,51 @@ import (
 	loggingconfig "github.com/rancher/rancher/pkg/controllers/user/logging/config"
 )
 
-func IsAllLoggingDisable(clusterLoggingLister v3.ClusterLoggingLister, projectLoggingLister v3.ProjectLoggingLister) (bool, error) {
-	clusterLoggings, err := clusterLoggingLister.List(loggingconfig.LoggingNamespace, labels.NewSelector())
+func IsAllLoggingDisable(clusterLoggingLister v3.ClusterLoggingLister, projectLoggingLister v3.ProjectLoggingLister, currentCL *v3.ClusterLogging, currentPL *v3.ProjectLogging) (bool, error) {
+	clusterLoggings, err := clusterLoggingLister.List("", labels.NewSelector())
 	if err != nil {
 		return false, err
 	}
 
-	projectLoggings, err := projectLoggingLister.List(loggingconfig.LoggingNamespace, labels.NewSelector())
+	projectLoggings, err := projectLoggingLister.List("", labels.NewSelector())
 	if err != nil {
 		return false, err
 	}
-	return len(clusterLoggings) == 0 && len(projectLoggings) == 0, nil
 
+	if len(clusterLoggings) == 0 && len(projectLoggings) == 0 {
+		return true, nil
+	}
+
+	for _, v := range clusterLoggings {
+		if currentCL != nil && v.Name == currentCL.Name {
+			v = currentCL
+		}
+		logging, _, err := GetWrapConfig(v.Spec.ElasticsearchConfig, v.Spec.SplunkConfig, v.Spec.SyslogConfig, v.Spec.KafkaConfig, v.Spec.EmbeddedConfig)
+		if err != nil {
+			return false, err
+		}
+		if logging.CurrentTarget != "" {
+			return false, nil
+		}
+	}
+
+	for _, v := range projectLoggings {
+		if currentPL != nil && v.Name == currentPL.Name {
+			v = currentPL
+		}
+		logging, _, err := GetWrapConfig(v.Spec.ElasticsearchConfig, v.Spec.SplunkConfig, v.Spec.SyslogConfig, v.Spec.KafkaConfig, nil)
+		if err != nil {
+			return false, err
+		}
+		if logging.CurrentTarget != "" {
+			return false, nil
```

Comment:
why return, not continue?

Reply:
once we get a target it means we don't need to remove the logging related components